share/availability/full/reconstruction_test.go

```diff
@@ -79,79 +118,119 @@ func TestShareAvailable_OneFullNode(t *testing.T) {
 // source, full node must be able to finish the availability process started in
 // the beginning.
 func TestShareAvailable_ConnectedFullNodes(t *testing.T) {
```

Comment:
I wonder, of this test is a subset of TestShareAvailable_DisconnectedFullNodes? It looks like it is just omitting part, when disconnected Full nodes cannot reconstruct and tests when they gets connected. Am I missing something or we can keep only TestShareAvailable_DisconnectedFullNodes?

Reply:
Its not exactly a subset but pretty close. I would agree that its kinda not necessary, but defer to @Wondertan for a decision on whether to keep it or not. I personally don't think it needs to exist

Reply:
I am keeping it because it is exactly what we need for the BarelyRecoverable condition - it doesn't test unrecoverability with only half of the nodes